daemon/db-core.c

```diff
@@ -88,7 +88,7 @@ struct qubesdb_entry *qubesdb_insert(struct qubesdb *db, char *path) {
         if (!new_entry)
             return NULL;
 #ifndef WIN32
-        strncpy(new_entry->path, path, QDB_MAX_PATH);
+        strncpy(new_entry->path, path, QDB_MAX_PATH - 1);
```

Comment:
In this case, add final '\0' manually. Or better - change malloc to calloc to get memory initialized with zeros.

Reply:
@marmarek: you mean for https://github.com/QubesOS/qubes-core-agent-linux/pull/109?